applications/sdp/mspi/src/main.c

```diff
@@ -52,215 +57,494 @@ typedef struct __packed {
-		nrf_vpr_clic_int_pending_set(NRF_VPRCLIC, VEVIF_IRQN(HRT_VEVIF_IDX_WRITE_QUAD));
-	}
+	ll_prepare_transfer(xfer_modes.address, nrfe_mspi_dev_cfg.addr_length);
+
+	nrf_vpr_clic_int_pending_set(NRF_VPRCLIC, VEVIF_IRQN(HRT_VEVIF_IDX_WRITE));
 
 	/* Send data */
-	xfer_ll_params.ce_hold = false;
-	xfer_ll_params.word_size = 32;
-	/* TODO: this system needs to be fixed as for now,
-	 * there are problems when (data_length%4) != 0
+	xfer_ll_params.bit_order = HRT_BO_REVERSED_BYTE;
+	/* TODO: Jira ticket: NRFX-6876 use read buffer that is appended to packet in NRFE_MSPI_TXRX,
+	 * this is not possible now due to alignment problems
 	 */
-	xfer_ll_params.data_len = WORDS_FOR_BYTES(data_length);
-	xfer_ll_params.data_to_send = data_buffer + XFER_DATA_IDX;
+	xfer_ll_params.data = nrfe_mspi_xfer_packet.data_buf;
 
-	if (mspi_dev_configs.io_mode == MSPI_IO_MODE_SINGLE) {
-		nrf_vpr_clic_int_pending_set(NRF_VPRCLIC, VEVIF_IRQN(HRT_VEVIF_IDX_WRITE_SINGLE));
-	} else {
-		nrf_vpr_clic_int_pending_set(NRF_VPRCLIC, VEVIF_IRQN(HRT_VEVIF_IDX_WRITE_QUAD));
-	}
+	ll_prepare_transfer(xfer_modes.data, nrfe_mspi_xfer_packet.num_bytes * BITS_IN_BYTE);
+
+	nrf_vpr_clic_int_pending_set(NRF_VPRCLIC, VEVIF_IRQN(HRT_VEVIF_IDX_WRITE));
 }
 
-__attribute__((interrupt)) void hrt_handler_write_single(void)
+static void process_packet(const void *data, size_t len)
 {
-	xfer_ll_params.ce_enable_state = (mspi_dev_configs.ce_polarity == MSPI_CE_ACTIVE_HIGH);
+	(void)len;
+	uint16_t dir;
+	uint16_t out;
+	nrfe_mspi_flpr_response_t response;
+	uint8_t opcode = *(uint8_t *)data;
 
-	write_single_by_word(xfer_ll_params);
-}
+	switch (opcode) {
+	case NRFE_MSPI_CONFIG_PINS: {
+		nrfe_mspi_pinctrl_soc_pin_t *pins_cfg = (nrfe_mspi_pinctrl_soc_pin_t *)data;
 
-__attribute__((interrupt)) void hrt_handler_write_quad(void)
-{
-	xfer_ll_params.ce_enable_state = (mspi_dev_configs.ce_polarity == MSPI_CE_ACTIVE_HIGH);
+		response.opcode = pins_cfg->opcode;
+
+		ce_vios_count = 0;
+		data_vios_count = 0;
+		dir = 0;
+		out = 0;
+
+		for (uint8_t i = 0; i < NRFE_MSPI_PINS_MAX; i++) {
+			uint32_t psel = NRF_GET_PIN(pins_cfg->pin[i]);
+			uint32_t fun = NRF_GET_FUN(pins_cfg->pin[i]);
+
+			uint8_t pin_number = psel & PINCTL_PIN_NUM_MASK;
+
+			if (pin_number >= sizeof(pin_to_vio_map)) {
+				/* TODO: Jira ticket: NRFX-6875 error*/
+				return;
+			}
+
+			dir = BIT_SET_VALUE(dir, pin_to_vio_map[pin_number],
+					    VPRCSR_NORDIC_DIR_OUTPUT);
+
+			if ((fun >= NRF_FUN_SDP_MSPI_CS0) && (fun <= NRF_FUN_SDP_MSPI_CS4)) {
+
+				NRFX_ASSERT(pin_number < sizeof(pin_to_vio_map))
+				ce_vios[ce_vios_count] = pin_to_vio_map[pin_number];
+				ce_vios_count++;
+
+				/* TODO: Jira ticket: NRFX-6876 Get CE disabled states and set them,
+				 * they need to be passed from app
+				 */
+			} else if ((fun >= NRF_FUN_SDP_MSPI_DQ0) && (fun <= NRF_FUN_SDP_MSPI_DQ7)) {
+
+				NRFX_ASSERT(pin_number < sizeof(pin_to_vio_map))
+				data_vios[data_vios_count] = pin_to_vio_map[pin_number];
+				data_vios_count++;
+			}
```

Comment:
Lack of CLK pin assignment.

Reply:
clk pin is always the same so I take it directly from NRFE_MSPI_SCK_PIN_NUMBER